File: languages/java/src/main/java/com/bloock/sdk/entity/RsaEncrypter.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.RecordOuterClass;
import com.bloock.sdk.bridge.proto.RecordOuterClass.EncryptionAlg;

public class RsaEncrypter implements Encrypter {
  EncryptionAlg alg;
  EncrypterArgs args;

  public RsaEncrypter(String publicKey) {
    this.alg = EncryptionAlg.RSA;
    this.args = new EncrypterArgs(publicKey);
  }

  @Override
  public RecordOuterClass.Encrypter toProto() {
    return RecordOuterClass.Encrypter.newBuilder()
        .setAlg(this.alg)
        .setArgs(this.args.toProto())
        .build();
  }
}

File: languages/php/Bloock/Entity/RsaEncrypter.php
<?php

namespace Bloock\Entity;

class RsaEncrypter implements Encrypter
{
    public $alg;
    public $args;

    public function __construct(string $publicKey) {
        $this->alg = \Bloock\EncryptionAlg::RSA;
        $this->args = new EncrypterArgs($publicKey);
    }

    public function toProto(): \Bloock\Encrypter {
        $p = new \Bloock\Encrypter();
        $p->setAlg($this->alg);
        $p->setArgs($this->args->toProto());
        return $p;
    }
}